aten/src/ATen/core/boxing/impl/boxing.h

```diff
@@ -194,34 +196,27 @@ struct BoxedKernelWrapper<
 };
 
 //
-// 3. signatures taking a single Tensor reference as their first argument,
-// and also returning one.
+// 3. in-place and legacy out-of-place ops take a single non-const Tensor
+// reference as their first argument, and return it.
+//
+// Note: all signatures matching this pattern are are assumed to be for such ops.
+// Because of this, the generated BoxedKernelWrapper specializations simply
+// return the in-place argument.
 //
-// Note that the passed kernels are assumed to be for inplace/outplace ops,
-// and the generated BoxedKernelWrapper specializations will simply return
-// the initial argument.
+// TODO update comment when legacy out-of-place signatures no longer need
+// to be supported, due to hacky_wrapper reordering
 //
 
 template <class... OtherArgs>
 struct BoxedKernelWrapper<
   at::Tensor&(at::Tensor&, OtherArgs...),
```

Comment:
that means this specialization only catches single-out-arg ops, not inplace ops. You might also have to rename test cases that you currently call "inplace".